flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreScanImpl.java

```diff
@@ -54,12 +55,13 @@
     private final ManifestFile.Factory manifestFileFactory;
     private final ManifestList manifestList;
 
-    private Long snapshotId;
-    private List<ManifestFileMeta> manifests;
     private Predicate partitionFilter;
     private Predicate keyFilter;
     private Predicate valueFilter;
-    private Integer bucket;
+
+    private Long specificSnapshotId = null;
```

Comment:
Nit: towards naming, personally, I prefer `specifiedSnapshotId` over `specificSanpshotId` cause **specified** sounds like somebody/the caller is specifying this value from outside (which is more suitable, I think) while `specific` sounds like the param is special.